Example/IntegrationTests/XPlatform/Web3Wallet/XPlatformW3WTests.swift

```diff
@@ -0,0 +1,144 @@
+
+        let relayClient = RelayClientFactory.create(
+            relayHost: InputConfig.relayHost,
+            projectId: InputConfig.projectId,
+            keyValueStorage: keyValueStorage,
+            keychainStorage: keychain,
+            socketFactory: DefaultSocketFactory(),
+            logger: relayLogger
+        )
+
+        let networkingClient = NetworkingClientFactory.create(
+            relayClient: relayClient,
+            logger: networkingLogger,
+            keychainStorage: keychain,
+            keyValueStorage: keyValueStorage)
+
+        let pairingClient = PairingClientFactory.create(
+            logger: pairingLogger,
+            keyValueStorage: keyValueStorage,
+            keychainStorage: keychain,
+            networkingClient: networkingClient)
+
+        let signClient = SignClientFactory.create(
+            metadata: AppMetadata(name: name, description: "", url: "", icons: [""]),
+            logger: signLogger,
+            keyValueStorage: keyValueStorage,
+            keychainStorage: keychain,
+            pairingClient: pairingClient,
+            networkingClient: networkingClient
+        )
+
+        let authClient = AuthClientFactory.create(
+            metadata: AppMetadata(name: name, description: "", url: "", icons: [""]),
+            projectId: InputConfig.projectId,
+            crypto: DefaultCryptoProvider(),
+            logger: authLogger,
+            keyValueStorage: keyValueStorage,
+            keychainStorage: keychain,
+            networkingClient: networkingClient,
+            pairingRegisterer: pairingClient,
+            iatProvider: DefaultIATProvider())
+
+        w3wClient = Web3WalletClientFactory.create(
+            authClient: authClient,
+            signClient: signClient,
+            pairingClient: pairingClient,
+            echoClient: EchoClientMock())
+    }
+
+    func testSessionSettle() async throws {
+
+        let expectation = expectation(description: "session settled")
+
+        w3wClient.sessionProposalPublisher
+            .sink { [unowned self] (proposal, _) in
+                Task(priority: .high) {
+                    let sessionNamespaces = SessionNamespace.make(toRespond: proposal.requiredNamespaces)
+                    try await w3wClient.approve(proposalId: proposal.id, namespaces: sessionNamespaces)
+                }
+            }
+            .store(in: &publishers)
+
+        w3wClient.sessionSettlePublisher.sink { [unowned self] session in
+            Task {
+                var jsSession: JavaScriptAutoTestsAPI.Session?
+
+                while jsSession == nil {
+                    print("🎃 geting session")
+                    do {
+                        jsSession = try await javaScriptAutoTestsAPI.getSession(topic: session.topic)
+                    } catch {
+                        print("No session on JS client yet")
+                    }
+
+                    if jsSession == nil {
+                        sleep(1)
```

Comment:
This is ok for now, but in future, if we are going to have more tests using the JS API, can they implement something like long polling?

Reply:
👍 good idea, 100%, will speak with js team